std/algorithm.d

```diff
@@ -14540,13 +14540,12 @@ auto cartesianProduct(RR...)(RR ranges)
         }
         @property Result save()
         {
-            Result copy;
+            Result copy = this;
```

Comment:
I would have preferred if this was wrapped in a `static if (isNested!Result)` (I think that's right). Performing the copy conservatively like this is unnecessarily expensive for normal ranges.